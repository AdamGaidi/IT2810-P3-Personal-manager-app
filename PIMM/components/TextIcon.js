import React from "react";
import { Icon } from "expo";

import Colors from "../constants/Colors";

export default class TextIcon extends React.Component {
  render() {
    const { defaultColor, focusedColor, iconSize } = this.props;

    return (
      <Icon.Ionicons
        name={this.props.name}
        size={iconSize}
        style={{ marginBottom: -3 }}
        color={this.props.focused ? focusedColor : defaultColor}
      />
    );
  }
}

TextIcon.defaultProps = {
  defaultColor: Colors.tabIconDefault,
  focusedColor: Colors.tabIconSelected,
  iconSize: 32
};
