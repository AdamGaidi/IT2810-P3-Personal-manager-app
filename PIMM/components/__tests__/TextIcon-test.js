import React from "react";
import TextIcon from "../TextIcon";
import CrossPlatformIcon from "../TextIcon";

import renderer from "react-test-renderer";

it("TextIcon renders correctly without props", () => {
  const tree = renderer.create(<TextIcon />).toJSON();
  expect(tree).toMatchSnapshot();
});
