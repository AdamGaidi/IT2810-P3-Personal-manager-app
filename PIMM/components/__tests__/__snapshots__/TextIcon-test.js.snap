// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextIcon renders correctly without props 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
/>
`;
